scripts/globals/garrison.lua

```diff
@@ -513,165 +178,318 @@ local spawnNPCs = function(npc)
+            debugLog("State: Grant Loot")
+            debugPrintToPlayers(players, "Mission success")
+
+            messagePlayers(npc, players, ID.text.GARRISON_BASE + 36)
+            xi.garrison.handleLootRolls(xi.garrison.loot[zoneData.levelCap], zoneData.players)
+            zoneData.state = xi.garrison.state.ENDED
         end,
-    }
 
-    -- TODO: Check to see if current wave is complete
+        [xi.garrison.state.ENDED] = function()
+            debugLog("State: Ended")
 
-    -- TODO: Check to if all waves are completed
+            xi.garrison.stop(zone)
+        end,
+    }
 
-    if zoneData.continue then
-        npc:timer(2000, function(npcArg)
+    if zoneData.isRunning then
+        npc:timer(1000, function(npcArg)
             xi.garrison.tick(npcArg)
         end)
     end
 end
 
 xi.garrison.start = function(player, npc)
-    -- TODO: Write lockout information to player
-
-    local zone         = player:getZone()
-    local zoneData     = xi.garrison.data[zone:getID()]
-    zoneData.players   = {}
-    zoneData.npcs      = {}
-    zoneData.mobs      = {}
-    zoneData.state     = xi.garrison.state.INACTIVE
-    zoneData.continue  = true
-    zoneData.stateTime = os.time()
-
+    local zone           = player:getZone()
+    local zoneData       = xi.garrison.zoneData[zone:getID()]
+    zoneData.players     = {}
+    zoneData.npcs        = {}
+    zoneData.mobs        = {}
+    zoneData.state       = xi.garrison.state.SPAWN_NPCS
+    zoneData.isRunning    = true
+    zoneData.stateTime   = os.time()
+    zoneData.waveIndex   = 1
+    zoneData.groupIndex  = 1
+    zoneData.bossSpawned = false
+    -- First mob spawn takes xi.garrison.waves.delayBetweenGroups to start
+    zoneData.nextSpawnTime = os.time() + xi.garrison.waves.delayBetweenGroups
+    zoneData.endTime           = os.time() + xi.settings.main.GARRISON_TIME_LIMIT
+    zoneData.deadNPCCount      = 0
+    zoneData.deadMobCount      = 0
+    zoneData.despawnedMobCount = 0
+
+    -- Adds level cap / registers lockout for all players
+    -- We register lockout at the beginning and end, in case players DC
     for _, member in pairs(player:getAlliance()) do
-        addLevelCap(member, zoneData.levelCap)
-        table.insert(zoneData.players, member:getID())
+        if member:getZoneID() == player:getZoneID() then
+            xi.garrison.addLevelCap(member, zoneData.levelCap)
+            xi.garrison.savePlayerLockout(member)
+
+            table.insert(zoneData.players, member:getID())
+        end
     end
 
-    -- The starting NPC is the 'anchor' for all timers and logic for this Garrison. Even though they
+    -- The starting NPC is the 'anchor' for all timers and logic for this Garrison
     xi.garrison.tick(npc)
 end
 
+-- Stops and cleans up the current garrison event (if any) on the given zone
+-- Can be called externally from GM commands
+xi.garrison.stop = function(zone)
+    local zoneData       = xi.garrison.zoneData[zone:getID()]
```

Comment:
Is this meant to be aligned with something?

Reply:
hmm no. On that note, I need to figure out how to either lint lua locally or stop vscode from showing these, because they throw off my alignment:
<img width="464" alt="Screen Shot 2022-11-17 at 9 56 30 AM" src="https://user-images.githubusercontent.com/116968546/202521631-810b8bcd-c879-40ed-a256-3668faa4f9c0.png">

Any ideas on how?

Reply:
Done